timesketch/lib/analyzers/browser_timeframe.py

```diff
@@ -0,0 +1,203 @@
+        next_lower, _ = runs[i+1]
+        if (upper + 1) == (next_lower - 1):
+            hour_list.append(upper + 1)
+
+    hours = sorted(hour_list)
+    runs = get_runs(hour_list)
+
+    if len(runs) <= 2:
+        return hours
+    elif len_runs < len(runs):
+        return fix_gap_in_list(hours)
+
+    # Now we need to remove runs, we only need the first and last.
+    run_start = runs[0]
+    run_end = runs[-1]
+    hours = list(range(0, run_start[1] + 1))
+    hours.extend(range(run_end[0], run_end[1] + 1))
+
+    return sorted(hours)
+
+
+def get_active_hours(frame):
+    """Return a list of the hours with the most activity within a frame.
+
+    Args:
+        frame: a pandas DataFrame object that contains a datetime column.
+
+    Returns:
+        A tuple that contains three items:
+            1. list of hours where the most activity within the DataFrame
+            occurs.
+            2. the threshold used to determine what is considered to be
+            an active hour.
+            3. a DataFrame object containing the aggregation.
+
+    """
+    frame_count = frame[['datetime', 'hour']].groupby(
+        'hour', as_index=False).count()
+    frame_count['count'] = frame_count['datetime']
+    del frame_count['datetime']
+
+    stats = frame_count['count'].describe()
+
+    # We use the 75% value - mean of all counts as the "bar" to which we
+    # determine an hour to be active.
+    threshold = stats['75%'] - stats['mean']
+
+    threshold_filter = frame_count['count'] >= threshold
+    hours = list(frame_count[threshold_filter].hour.values)
+    hours = sorted(hours)
+
+    runs = get_runs(hours)
+
+    # There should either be a single run or at most two.
+    number_runs = len(runs)
+    if number_runs == 1:
+        return hours, threshold, frame_count
+
+    elif number_runs == 2 and runs[0][0] == 0:
+        # Two runs, first one starts at hour zero.
+        return hours, threshold, frame_count
+
+    return fix_gap_in_list(hours), threshold, frame_count
+
+
+class BrowserTimeframeSketchPlugin(interface.BaseSketchAnalyzer):
+    """Sketch analyzer for BrowserTimeframe."""
+
+    NAME = 'browser_timeframe'
+    DEPENDENCIES = frozenset()
+
+    def __init__(self, index_name, sketch_id):
+        """Initialize The Sketch Analyzer.
+
+        Args:
+            index_name: Elasticsearch index name
+            sketch_id: Sketch ID
+        """
+        self.index_name = index_name
+        super(BrowserTimeframeSketchPlugin, self).__init__(
+            index_name, sketch_id)
+
+    def run(self):
+        """Entry point for the analyzer.
+
+        Returns:
+            String with summary of the analyzer result
+        """
+        # TODO: Once we can identify user generated events this should be
+        # updated to include all user generated events instead of focusing
+        # solely on browser events.
+        query = 'source_short:"WEBHIST"'
+
+        return_fields = ['timestamp', 'url', 'tag', '__ts_emojis']
```

Comment:
You doesn't use the "__ts_emojis" anywhere so you can drop it from the returned fields.

Reply:
Aha! magic ;)